src/plots/plot.ts

```diff
@@ -489,7 +489,7 @@ export class Plot extends Component {
 
       dataset.data().forEach((datum: any, datasetIndex: number) => {
         let position = this._pixelPoint(datum, datasetIndex, dataset);
-        if (Utils.Math.isNaN(position.x) || Utils.Math.isNaN(position.y)) {
+        if (!Utils.Math.isValidNumber(position.x) || !Utils.Math.isValidNumber(position.y)) {
```

Comment:
Outside the scope of this refactor. Skip test (if failing), and make separate PR please.